src/arrayMethodJoin.js

```diff
@@ -6,6 +6,29 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (this[this.length - 1] === undefined) {
+      this[this.length - 1] = '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (separator !== undefined) {
+        result += this[i] + `${separator}`;
+      } else {
+        result += this[i] + ',';
+      }
```

Comment:
you do `result += this[i]` in both cases, better take it out of the `if` statement